### ISSUE TITLE
v4: Support redis cluster 7.x

### ISSUE THREAD

**Issue by salarali** [mediocregopher/radix#316]
When I try to use redis cluster 7.x, I get the following error:

```
radix fail: response returned from Conn: unmarshaling message off Conn: cannot decode resp array into string
```

Im guessing this because cluster slots command output is changed and that the cluster slots command is deprecated.

**Comment by mediocregopher** [comment]
Hello @salarali , can you provide a minimal code example which generates this error?

As far as I know the response from `CLUSTER SLOTS` is not changed at all, only a new command `CLUSTER SHARDS` has been added, so I don't expect anything to be broken from this.

**Comment by arthurkamwt10** [comment]
Sample code:
```
// github.com/mediocregopher/radix/v4
func main() {
	_, err := (radix.ClusterConfig{}).New(context.TODO(), []string{"127.0.0.1:7000", "127.0.0.1:7001", "127.0.0.1:7002"})
	if err != nil {
		log.Panic("radix fail:", err)
	}
	log.Println("radix ok")
}
```

Response from v7 cluster:
```
2022/06/01 13:11:08 radix fail:response returned from Conn: unmarshaling message off Conn: cannot decode resp array into string
```
v7 cluster slots:
```
127.0.0.1:7000> cluster slots
1) 1) (integer) 0
   2) (integer) 5460
   3) 1) "127.0.0.1"
      2) (integer) 7000
      3) "2059dbe24a3ecce58d920cba4e2193ad0a7c0891"
      4) (empty array)
2) 1) (integer) 5461
   2) (integer) 10922
   3) 1) "127.0.0.1"
      2) (integer) 7001
      3) "c922df5935b8149eaf75d520909b9204ec2b783e"
      4) (empty array)
3) 1) (integer) 10923
   2) (integer) 16383
   3) 1) "127.0.0.1"
      2) (integer) 7002
      3) "219d4b46a7252904adfe087a9a2e2a763538bf72"
      4) (empty array)
```

v6 cluster slots:
```
127.0.0.1:7000> cluster slots
1) 1) (integer) 0
   2) (integer) 5460
   3) 1) "127.0.0.1"
      2) (integer) 7000
      3) "0d9f891506bd8762cbf10285239c75bacef86b0b"
2) 1) (integer) 5461
   2) (integer) 10922
   3) 1) "127.0.0.1"
      2) (integer) 7001
      3) "6027cf70b8e6e88d43e7456c1a442d012d853d11"
3) 1) (integer) 10923
   2) (integer) 16383
   3) 1) "127.0.0.1"
      2) (integer) 7002
      3) "9431c6ec0caeb481b123f84e894e47b4de365d03"
```

There are extra `(empty array)` entries in the v7 slots, which may have been the issue?

**Comment by salarali** [comment]
There is another field added to the output of `cluster slots`
 > A map of additional networking metadata

https://redis.io/commands/cluster-slots/


Also, `cluster slots` is considered deprecated starting 7.x and replaced by `cluster-shards`

**Comment by mediocregopher** [comment]
Ah I see, the new field isn't unmarshalable into a string.

@salarali I pushed a potential fix to a branch at #320 , if you could check that out and let me know if it works for you I'd very much appreciate it :pray:
